addon/session-stores/cookie.js

```diff
@@ -99,28 +98,34 @@ export default BaseStore.extend({
 
     @method persist
     @param {Object} data The data to persist
+    @return {Ember.RSVP.Promise} The promise object persisting the data in the store.
     @public
   */
   persist(data) {
-    data           = JSON.stringify(data || {});
-    let expiration = this._calculateExpirationTime();
-    this._write(data, expiration);
-    this._lastData = this.restore();
+    return new RSVP.Promise((resolve, reject) => {
+      data           = JSON.stringify(data || {});
+      let expiration = this._calculateExpirationTime();
+      this._write(data, expiration);
+      this.restore().then((restoredContent) => {
+        this._lastData = restoredContent;
+        resolve();
+      }, reject);
```

Comment:
That's true, however something on the store is not correct in that case. Changed it to `resolve` anyway